### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by stakeworks** [bifrost-finance/bifrost#620]
Ubuntu: 20.04 LTS
8 cores
30 GB RAM
bifrost 0.9.44-7af39eb

After upgrade to bifrost 0.9.44-7af39eb memory usage increases a lot! (the memory issue also happened after upgrade to robonomics 2.0.0-17b47a4bd7c)

![image](https://user-images.githubusercontent.com/19406838/173845471-0a290325-2f4b-451e-bbc3-a8cc16eb8abc.png)

![image](https://user-images.githubusercontent.com/19406838/173845658-1e5b8e63-b901-44d4-a51f-7aeb0311d985.png)

Don't know if it's related but also getting these messages:

Labels
alertname = UnboundedChannelPersistentlyLarge
chain = bifrost
entity = mpsc_finality_notification_stream
instance = localhost:19635
job = bifrost-kusama-cmn12
monitor = CMN12
severity = warning
Annotations
message = Channel mpsc_finality_notification_stream on node localhost:19635 contains more than 200 items for more than 5 minutes. Node might be frozen.

**Comment by Akagi201** [comment]
It seems that it is releated to this issue https://github.com/paritytech/substrate/issues/11604

**Comment by koute** [comment]
If you have the following at the start of your logs:

```
[Relaychain] BEEFY is still experimental, usage on a production network is discouraged.
```

...*and* you're not actually using BEEFY then it's most likely the same issue.

If so here's a fix you can test out: https://github.com/paritytech/substrate/pull/11694

**Comment by stakeworks** [comment]
Looks like it's been solved in version 0.9.52, memory usage is still normal after an hour. Will keep an eye on it for the next days.

![image](https://user-images.githubusercontent.com/19406838/181070261-446ae441-31c9-4b97-8f37-ae04ec1cfe97.png)

**Comment by ark930** [comment]
Fixed by the version bifrost-v0.9.52: https://github.com/bifrost-finance/bifrost/releases/tag/bifrost-v0.9.52-1
